mily/widgets.py

```diff
@@ -66,6 +66,132 @@ def set_default(self, v):
+    are common for all ``mily`` widgets to make the consumer code easier. It
+    also adds a ``name`` attribute and ``__init__`` argument.
+
+    Parameters
+    ----------
+    args: various
+        args passed to ``PyQt5.QtWidgets.QCheckBox.__init__(...)``.
+    kwargs: various
+        kwargs passed to ``pyQt5.QtWidgets.QCheckBox.__init__(...)``.
+    '''
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+        self._name = name
+
+    def get_parameters(self):
+        '''returns a ``{name: isChecked}`` dictionary. '''
+        return {self._name: self.isChecked()}
+
+    def set_default(self, checkState):
+        '''set ``self.isChecked`` to ``checkState``.'''
+        if checkState is not None:
+            self.setChecked(checkState)
+
+
+class MSelector(QtWidgets.QGroupBox):
+    '''A widget that displays a list of objects with 'checkboxes'.
+
+    This adds ``get_parameters(self)`` and ``set_default(self,v)`` methods that
+    are common for all ``mily`` widgets to make the consumer code easier. It
+    also adds a ``name`` attribute and ``__init__`` argument. It results in a
+    vertical list of objects (defined in the 'items' arg) with a corresponding
+    checkbox for each item. It returns a list of checked items.
+
+    Parameters
+    ----------
+    name: str
+        name of the widget to be stored in ``self._name``
+    option_list : list
+        list of items to include checkboxes for.
+    vertical : bool, optional
+        optional boolean that indicates if the list should be displayed
+        vertically (True) or horizontally (False). default is True.
+    kwargs: various
+        kwargs passed to ``pyQt5.QtWidgets.QComboBox.__init__(...)``.
+    '''
+
+    def __init__(self, name, option_list, vertical=True, **kwargs):
+        self._name = name
+        super().__init__(title='', **kwargs)
+        self.button_group = QtWidgets.QButtonGroup()
+        self.button_group.setExclusive(False)
+        if vertical:
+            MainLayout = QtWidgets.QVBoxLayout()
+        else:
+            MainLayout = QtWidgets.QHBoxLayout()
+        self.setLayout(MainLayout)
+        for item in option_list:
+            item_name = getattr(item, 'name', str(item))
+            button = QtWidgets.QCheckBox(item_name, **kwargs)
+            setattr(button, 'item', item)
+            self.button_group.addButton(button)
+            MainLayout.addWidget(button)
+
+        # sets the layout width and height to fit inclosed widgets
+        self.layout().setSizeConstraint(QtWidgets.QLayout.SetFixedSize)
+        # ensure that the background is auto-filled
+        self.setAutoFillBackground(True)
+
+    def get_parameters(self):
+        item_list = [b.item
+                     for b in self.button_group.buttons()
+                     if b.isChecked()]
+
+        return {self._name: item_list}
+
+    def set_default(self, value):
```

Comment:
Why not to name the `value` as `dets_list` or something less generic?

Reply:
This is a non generic widget, i.e. it can be used to select one or more items from _any_ list not just a list of detectors. Therefore it would be very confusing if this was labeled ``dets_list`` and this widget was being used on a list of strings.

As a compromise I could change it to ``item_list`` as s used in the ``get_parameters`` method above. Does this resolve your issue with 'value'? The other option would be to change both here and in 'get_parameters' to use 'parameters'.

Reply:
fixed, thanks @mrakitin